corehq/motech/repeaters/models.py

```diff
@@ -1157,9 +1157,9 @@ def already_processed():
         task = retry_process_repeat_record if is_retry else process_repeat_record
 
         if fire_synchronously:
-            task(self)
+            task(self._id)
```

Comment:
I thought about having these tasks accept both an id or object to optimize the synchronous flow, and am open to it. It felt clunky, but if it is worth it I'm happy to make that change.

Reply:
I noticed that `retry_process_repeat_record()` and `process_repeat_record()` are identical, and they both call `_process_repeat_record()`.

What if we ...
1. drop `process_repeat_record()`,
2. rename `_process_repeat_record()` to `process_repeat_record()`,
3. and rename `retry_process_repeat_record()` to `process_repeat_record_async()`?

Then `process_repeat_record_async()` takes an ID, and `process_repeat_record()` is called synchronously with a RepeatRecord instance?

(I suspect the answer might be "no", and probably has something to do with the reason behind why  `retry_process_repeat_record()` and `process_repeat_record()` both exist. (Monitoring?))

I'm not certain about the speed considerations. But I am very much looking forward to the day this is on SQL, and we can stop trying to send repeat records whose repeaters are paused, or whose remote endpoints are offline. Then these speed considerations will become irrelevant.

Reply:
Confirming they are separate for monitoring purposes. I added some docs to those methods to make it more clear so thanks for flagging.